Validation/RecoTrack/plugins/TrackingNtuple.cc

```diff
@@ -3260,29 +3200,13 @@ void TrackingNtuple::fillSeeds(const edm::Event& iEvent,
       see_stateTrajGlbPx.push_back(stateGlobal.momentum().x());
       see_stateTrajGlbPy.push_back(stateGlobal.momentum().y());
       see_stateTrajGlbPz.push_back(stateGlobal.momentum().z());
-      if (addSeedCartesianCov_) {
-        auto const& stateCcov = tsos.cartesianError().matrix();
-        see_stateCcov00.push_back(stateCcov[0][0]);
-        see_stateCcov01.push_back(stateCcov[0][1]);
-        see_stateCcov02.push_back(stateCcov[0][2]);
-        see_stateCcov03.push_back(stateCcov[0][3]);
-        see_stateCcov04.push_back(stateCcov[0][4]);
-        see_stateCcov05.push_back(stateCcov[0][5]);
-        see_stateCcov11.push_back(stateCcov[1][1]);
-        see_stateCcov12.push_back(stateCcov[1][2]);
-        see_stateCcov13.push_back(stateCcov[1][3]);
-        see_stateCcov14.push_back(stateCcov[1][4]);
-        see_stateCcov15.push_back(stateCcov[1][5]);
-        see_stateCcov22.push_back(stateCcov[2][2]);
-        see_stateCcov23.push_back(stateCcov[2][3]);
-        see_stateCcov24.push_back(stateCcov[2][4]);
-        see_stateCcov25.push_back(stateCcov[2][5]);
-        see_stateCcov33.push_back(stateCcov[3][3]);
-        see_stateCcov34.push_back(stateCcov[3][4]);
-        see_stateCcov35.push_back(stateCcov[3][5]);
-        see_stateCcov44.push_back(stateCcov[4][4]);
-        see_stateCcov45.push_back(stateCcov[4][5]);
-        see_stateCcov55.push_back(stateCcov[5][5]);
+      if (addSeedCurvCov_) {
+        auto const& stateCcov = tsos.curvilinearError().matrix();
+        std::vector<float> cov(15);
+        auto covP = cov.begin();
+        for (auto const val : stateCcov)
+          *(covP++) = val;//row-major
+        see_stateCurvCov.push_back(cov);
```

Comment:
In principle
```suggestion
        see_stateCurvCov.emplace_back(std::move(cov));
```
would avoid a copy.

Reply:
doesn't this get optimized to the same result (especially considering the few-line scope)?

Reply:
Godbolt shows difference (fewer assembly lines with `std::move`)), and without `std::move()` standard requires to make a copy (IIUC). `push_back` vs. `emplace_back` showed no difference.